backend/plonk/bn254/solidity.go

```diff
@@ -25,123 +25,141 @@ pragma experimental ABIEncoderV2;
 library Utils {
 
   uint256 constant r_mod = 21888242871839275222246405745257275088548364400416034343698204186575808495617;
+  uint8 private constant zero = 0;
+  uint8 private constant lenInBytes = 48;
+  uint8 private constant sizeDomain = 11; // size of dst
 
   /**
   * @dev ExpandMsgXmd expands msg to a slice of lenInBytes bytes.
   *      https://tools.ietf.org/html/draft-irtf-cfrg-hash-to-curve-06#section-5
   *      https://tools.ietf.org/html/rfc8017#section-4.1 (I2OSP/O2ISP)
   */
-  function expand_msg(uint256 x, uint256 y) public pure returns(uint8[48] memory res){
-  
-      string memory dst = "BSB22-Plonk";
-
-      //uint8[64] memory pad; // 64 is sha256 block size.
-      // sha256(pad || msg || (0 || 48 || 0) || dst || 11)
-      bytes memory tmp;
-      uint8 zero = 0;
-      uint8 lenInBytes = 48;
-      uint8 sizeDomain = 11; // size of dst
-      
-      for (uint i=0; i<64; i++){
-          tmp = abi.encodePacked(tmp, zero);
+  function expand_msg(uint256 x, uint256 y) public pure returns (uint8[48] memory res) {
+    string memory dst = "BSB22-Plonk";
```

Comment:
Is it possible to make a global constant out of "BSB22-Plonk" also ?

Reply:
indeed it is